storage/remote/azuread/azuread.go

```diff
@@ -84,18 +100,47 @@ func (c *AzureADConfig) Validate() error {
 		return fmt.Errorf("must provide a cloud in the Azure AD config")
 	}
 
-	if c.ManagedIdentity == nil {
-		return fmt.Errorf("must provide an Azure Managed Identity in the Azure AD config")
+	if c.ManagedIdentity == nil && c.OAuth == nil {
+		return fmt.Errorf("must provide an Azure Managed Identity or Azure OAuth in the Azure AD config")
 	}
 
-	if c.ManagedIdentity.ClientID == "" {
-		return fmt.Errorf("must provide an Azure Managed Identity client_id in the Azure AD config")
+	if c.ManagedIdentity != nil && c.OAuth != nil {
+		return fmt.Errorf("cannot provide both Azure Managed Identity and Azure OAuth in the Azure AD config")
 	}
 
-	_, err := uuid.Parse(c.ManagedIdentity.ClientID)
-	if err != nil {
-		return fmt.Errorf("the provided Azure Managed Identity client_id provided is invalid")
+	if c.ManagedIdentity != nil {
+		if c.ManagedIdentity.ClientID == "" {
+			return fmt.Errorf("must provide an Azure Managed Identity client_id in the Azure AD config")
+		}
+
+		_, err := uuid.Parse(c.ManagedIdentity.ClientID)
+		if err != nil {
+			return fmt.Errorf("the provided Azure Managed Identity client_id provided is invalid")
+		}
+	}
+
+	if c.OAuth != nil {
+		if c.OAuth.ClientID == "" {
+			return fmt.Errorf("must provide an Azure OAuth client_id in the Azure AD config")
+		}
+		if c.OAuth.ClientSecret == "" {
+			return fmt.Errorf("must provide an Azure OAuth client_secret in the Azure AD config")
+		}
+		if c.OAuth.TenantID == "" {
+			return fmt.Errorf("must provide an Azure OAuth tenant_id in the Azure AD config")
+		}
+
+		var err error
+		_, err = uuid.Parse(c.OAuth.ClientID)
+		if err != nil {
+			return fmt.Errorf("the provided Azure OAuth client_id provided is invalid")
+		}
+		_, err = uuid.Parse(c.OAuth.TenantID)
```

Comment:
Are we sure that the tenant ID has to be a uuid? In the documentation [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#first-case-access-token-request-with-a-shared-secret) it says that "The directory tenant the application plans to operate against, **in GUID or domain-name format.**"

Reply:
I looked into and you are right. Though not commonly used, the tenant ID could be domain-name format as well. I updated this check to mirror the check the underlying library uses to validate the tenant ID as shown here:
https://github.com/Azure/azure-sdk-for-go/blob/sdk/azidentity/v1.3.0/sdk/azidentity/azidentity.go#L125:L131